package.json

```diff
@@ -98,7 +95,7 @@
     "mock-mail-adapter": "file:spec/dependencies/mock-mail-adapter",
     "mongodb-runner": "4.8.1",
     "mongodb-version-list": "1.0.0",
-    "node-fetch": "3.1.1",
```

Comment:
Why do we need to downgrade it?

Reply:
I don't think we should. The package is used only by a GraphQL test file and there seems to an incompatibility, see https://github.com/parse-community/parse-server/issues/7978. I have disabled the 2 tests for now, it's an open issue to replace node-fetch.

Reply:
> I have disabled the 2 tests for now

I think it's not a good idea, Multipart File Upload is an important feature of the GQL API. It could definitely break many parse-servers deployment, this feature MUST be validated by the CI @mtrezza

Reply:
If the feature must be validated then I suggest we fix the node-fetch issue.

Reply:
@davimacedo  `node-fetch` is only used in the ParseGraphQL.Server test file, it's safe to downgrade to 2.X until a PR could be sent with the 3.X support or a complete refactor with another lib.

Reply:
> Remember, 2.x has been upgraded to 3.x. due to a security vulnerability.

@mtrezza I understand perfectly, but here the potential security issue only affects the parse server test localhost test suite. I do not agree to disable tests related to important features used for an internal dev package version issue with low-security risk, since this package is only a dev dep.

I'm may be wrong but the breaking change from 2.X to 3.X on node fetch is related to the switch to ESM. I discussed this also with The Guild.

I see that the upgrade was created  here https://github.com/parse-community/parse-server/pull/7782

And also a security update was also released on the 2.X: https://github.com/node-fetch/node-fetch/releases/tag/v2.6.7

<img width="811" alt="Capture d’écran 2022-05-06 à 20 41 28" src="https://user-images.githubusercontent.com/27959372/167198798-23814b53-383e-466b-8d56-7d92f5d0a2f6.png">

The 2.X has also a LTS it seems. @mtrezza

Reply:
> since this package is only a dev dep.

We have seen major security vulnerabilities in the past due to dev dependencies. The package is there and we cannot know what else it will be used for in a CI/CD pipeline.

> The 2.X has also a LTS it seems

Until when?

Reply:
form npm page

```
node-fetch from v3 is an ESM-only module - you are not able to import it with require().

If you cannot switch to ESM, please use v2 which remains compatible with CommonJS. Critical bug fixes will continue to be published for v2.
```

Not mentioned, an issue could be opened on the node-fetch repo.

Anyway this is just my opinion, I don't think it's a reasonable solution to disable the tests compared to the current situation. in any case I will keep node-fetch 2.X on my fork to ensure that this feature works. I don't think I'll investigate further on my side a PR to switch back to node 3.X or another fetch package.

Reply:
Yes, I think we could just ask them for a LTS end date and downgrade to 2.x. After that date we'll have to upgrade anyway, so we are just moving the debt.

Reply:
See https://github.com/node-fetch/node-fetch/discussions/1553#discussioncomment-2703103.

They already released beta versions of 4.x, plus people are likely switching to Node's built-in `fetch` which is available from Node 18. I think we cannot be sure how well maintained 2.x will stay. Their maintenance model seems to be more community driven than policy driven, so it's hard to tell.

Maybe let's try to go with the compromise:
- Stay on 3.x and run the tests only for Node <17
- Do not run the tests on Node 17, which is an odd-intermediate version where LTS ends 2022-06 anyway
- Over time, create a PR to run the tests on Node 18 using Node's the built-in `fetch`

Reply:
@ardatan I tried the switch to cross-undici-fetch but new fails appeared. The switch from node-fetch is out of the scope of this PR from my point of view.

Feel free to open a new PR to switch replace node-fetch by cross-undici-fetch you will maybe able to fix fails more easily and quickly than me. 🙂